code/modules/power/rbmk/rbmk_monitoring.dm

```diff
@@ -0,0 +1,150 @@
+	var/next_stat_interval = 0
+	var/list/kpaData = list()
+	var/list/powerData = list()
+	var/list/tempInputData = list()
+	var/list/tempOutputdata = list()
+	var/list/reactors
+	var/obj/machinery/atmospherics/components/unary/rbmk/core/reactor // Currently selected RBMK Reactor.
+
+/datum/computer_file/program/nuclear_monitor/Destroy()
+	clear_signals()
+	reactor = null
+	return ..()
+
+/datum/computer_file/program/nuclear_monitor/process_tick()
+	..()
+	if(!reactor || !active)
+		return FALSE
+	var/new_status = get_status()
+	if(last_status != new_status)
+		last_status = new_status
+		ui_header = "smmon_[last_status].gif"
+		program_icon_state = "smmon_[last_status]"
+		if(istype(computer))
+			computer.update_icon()
+	if(world.time >= next_stat_interval)
+		next_stat_interval = world.time + 1 SECONDS //You only get a slow tick.
+		kpaData += (reactor) ? reactor.pressure : 0
+		if(kpaData.len > 100) //Only lets you track over a certain timeframe.
+			kpaData.Cut(1, 2)
+		powerData += (reactor) ? reactor.power*10 : 0 //We scale up the figure for a consistent:tm: scale
+		if(powerData.len > 100) //Only lets you track over a certain timeframe.
+			powerData.Cut(1, 2)
+		tempInputData += (reactor) ? reactor.last_coolant_temperature : 0 //We scale up the figure for a consistent:tm: scale
+		if(tempInputData.len > 100) //Only lets you track over a certain timeframe.
+			tempInputData.Cut(1, 2)
+		tempOutputdata += (reactor) ? reactor.last_output_temperature : 0 //We scale up the figure for a consistent:tm: scale
+		if(tempOutputdata.len > 100) //Only lets you track over a certain timeframe.
+			tempOutputdata.Cut(1, 2)
+
+/datum/computer_file/program/nuclear_monitor/on_start(mob/living/user)
+	. = ..(user)
+	//No reactor? Go find one then.
+	if(!reactor)
+		for(var/obj/machinery/atmospherics/components/unary/rbmk/core/reactor_core in GLOB.machines)
+			if(user.get_virtual_z_level() == reactor_core.get_virtual_z_level())
+				reactor = reactor_core
+				break
+	active = TRUE
+
+/datum/computer_file/program/nuclear_monitor/kill_program(forced = FALSE)
+	active = FALSE
+	..()
+
+/datum/computer_file/program/nuclear_monitor/ui_data()
+	var/list/data = list()
+	data["integrity"] = reactor ? reactor.get_integrity_percent() : 100
+	data["powerData"] = powerData
+	data["kpaData"] = kpaData
+	data["tempInputData"] = tempInputData
+	data["tempOutputdata"] = tempOutputdata
+	data["coolantInput"] = reactor ? reactor.last_coolant_temperature : 0
+	data["coolantOutput"] = reactor ? reactor.last_output_temperature : 0
+	data["power"] = reactor ? reactor.power : 0
+	data["kpa"] = reactor ? reactor.pressure : 0
+	return data
+
+/datum/computer_file/program/nuclear_monitor/ui_act(action, params)
+	if(..())
+		return TRUE
+
+	switch(action)
+		if("swap_reactor")
+			var/list/choices = list()
+			for(var/obj/machinery/atmospherics/components/unary/rbmk/core/reactor_core in GLOB.machines)
+				if(usr.get_virtual_z_level() != reactor_core.get_virtual_z_level())
```

Comment:
```suggestion
			var/user_z_level = usr.get_virtual_z_level()
			for(var/obj/machinery/atmospherics/components/unary/rbmk/core/reactor_core in GLOB.machines)
				if(!compare_z(user_z_level, reactor_core.get_virtual_z_level())
```
I am not sure if doing `usr` is right here, but whatever. it won't be breaking the game...